waypoint_generator/src/waypoint_generator/waypoint_generator.cpp

```diff
@@ -516,10 +521,27 @@ void WaypointGenerator::new_route_callback()
   ROS_DEBUG("Generating final waypoint message.");
   autoware_msgs::LaneArray waypoint_msg;
   waypoint_msg = this->generate_lane_array_message(final_speeds, orientations, centerline);
+  waypoint_msg = this->downsample_waypoints(waypoint_msg, _config._downsample_ratio);
 
   ROS_DEBUG_STREAM("Finished processing route.");
 
   _waypoint_publisher(waypoint_msg);
   ROS_DEBUG_STREAM("Published waypoints list!");
 }
+
+autoware_msgs::LaneArray WaypointGenerator::downsample_waypoints(autoware_msgs::LaneArray waypoints, int ratio) const
+{
+  autoware_msgs::LaneArray downsampled{waypoints};
+
+  for (int i = 0; i < downsampled.lanes.size(); i++) {
+    for (int j = 0; j < downsampled.lanes[i].waypoints.size(); j++) {
+      if (j % ratio != 0) {
+        downsampled.lanes[i].waypoints.erase(downsampled.lanes[i].waypoints.begin() + j);
```

Comment:
Are you confident this is a safe loop. I think normally when changing the size of the vector while iterating over it you want to use iterator syntax. 

https://stackoverflow.com/questions/9927163/erase-element-in-vector-while-iterating-the-same-vector